<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>sousvide</class>
 <widget class="QMainWindow" name="sousvide">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>701</width>
    <height>562</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>sousvide</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QGroupBox" name="groupTemperature">
        <property name="title">
         <string>Temperatures</string>
        </property>
        <layout class="QFormLayout" name="formLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Input</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLabel" name="T_input">
           <property name="text">
            <string>69 °C</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Smoothed</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLabel" name="T_smooth">
           <property name="text">
            <string>68 °C</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>Setpoint</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLabel" name="T_setpoint">
           <property name="text">
            <string>70 °C</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_5">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QGroupBox" name="groupTimer">
        <property name="title">
         <string>Timer</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QLabel" name="timeRemaining">
           <property name="font">
            <font>
             <family>Droid Sans Mono</family>
             <pointsize>31</pointsize>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>00:00:00</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QGroupBox" name="groupConstants">
        <property name="title">
         <string>PID Constants</string>
        </property>
        <layout class="QFormLayout" name="formLayout_3">
         <item row="0" column="0">
          <widget class="QLabel" name="label_15">
           <property name="text">
            <string>P</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLabel" name="const_P">
           <property name="text">
            <string>1.0</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_14">
           <property name="text">
            <string>I</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLabel" name="const_I">
           <property name="text">
            <string>0.01</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_18">
           <property name="text">
            <string>D</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLabel" name="const_D">
           <property name="text">
            <string>0.0</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupState">
        <property name="title">
         <string>State</string>
        </property>
        <layout class="QFormLayout" name="formLayout_2">
         <item row="0" column="0">
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>Integral</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLabel" name="integral">
           <property name="text">
            <string>32</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_9">
           <property name="text">
            <string>Derivative</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLabel" name="derivative">
           <property name="text">
            <string>52</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_11">
           <property name="text">
            <string>Output</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLabel" name="output">
           <property name="text">
            <string>127</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QCustomPlot" name="plot" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>701</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionClearGraph"/>
    <addaction name="actionSaveInto"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menu_Help">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuCooking">
    <property name="title">
     <string>&amp;Cooking</string>
    </property>
    <addaction name="actionSetSetpointTemperature"/>
    <addaction name="actionStartTimer"/>
    <addaction name="actionStopTimer"/>
    <addaction name="separator"/>
    <addaction name="actionConnect"/>
    <addaction name="actionResetController"/>
    <addaction name="actionModifyConstants"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menuCooking"/>
   <addaction name="menu_Help"/>
  </widget>
  <action name="actionQuit">
   <property name="icon">
    <iconset resource="../icons.qrc">
     <normaloff>:/icons/icons/cross.png</normaloff>:/icons/icons/cross.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Quit</string>
   </property>
  </action>
  <action name="actionSaveInto">
   <property name="icon">
    <iconset resource="../icons.qrc">
     <normaloff>:/icons/icons/disk.png</normaloff>:/icons/icons/disk.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Save Into</string>
   </property>
  </action>
  <action name="actionClearGraph">
   <property name="icon">
    <iconset resource="../icons.qrc">
     <normaloff>:/icons/icons/chart_line_edit.png</normaloff>:/icons/icons/chart_line_edit.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Clear Graph</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="icon">
    <iconset resource="../icons.qrc">
     <normaloff>:/icons/icons/tux.png</normaloff>:/icons/icons/tux.png</iconset>
   </property>
   <property name="text">
    <string>&amp;About</string>
   </property>
  </action>
  <action name="actionConnect">
   <property name="icon">
    <iconset resource="../icons.qrc">
     <normaloff>:/icons/icons/connect.png</normaloff>:/icons/icons/connect.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Connect</string>
   </property>
   <property name="toolTip">
    <string>Connect</string>
   </property>
  </action>
  <action name="actionStartTimer">
   <property name="icon">
    <iconset resource="../icons.qrc">
     <normaloff>:/icons/icons/clock.png</normaloff>:/icons/icons/clock.png</iconset>
   </property>
   <property name="text">
    <string>Start &amp;Timer</string>
   </property>
  </action>
  <action name="actionResetController">
   <property name="icon">
    <iconset resource="../icons.qrc">
     <normaloff>:/icons/icons/cancel.png</normaloff>:/icons/icons/cancel.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Reset Controller State</string>
   </property>
  </action>
  <action name="actionSetSetpointTemperature">
   <property name="icon">
    <iconset resource="../icons.qrc">
     <normaloff>:/icons/icons/wand.png</normaloff>:/icons/icons/wand.png</iconset>
   </property>
   <property name="text">
    <string>Set &amp;Setpoint Temperature</string>
   </property>
  </action>
  <action name="actionModifyConstants">
   <property name="icon">
    <iconset resource="../icons.qrc">
     <normaloff>:/icons/icons/cog_edit.png</normaloff>:/icons/icons/cog_edit.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Modify PID Constants</string>
   </property>
  </action>
  <action name="actionStopTimer">
   <property name="icon">
    <iconset resource="../icons.qrc">
     <normaloff>:/icons/icons/clock_stop.png</normaloff>:/icons/icons/clock_stop.png</iconset>
   </property>
   <property name="text">
    <string>Stop T&amp;imer</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>sousvide</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>350</x>
     <y>280</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
